### PR TITLE
BUGFIX: Fix double slash after base uri

### LINKED ISSUE #89: RedirectFinisher causes double slash
*Steps to reproduce:*

* Create a form with a redirectFinisher pointing to a document node (I used the form builder package and created a node based form)
* Publish the form
* Fill out the form in the frontend and submit it

*Expected behavior*
You get redirected to the destination node and the uri is fine.

*Actual behaviour*
The uri contains a double-slash.

I think the reason is found here: https://github.com/neos/form/blame/3d4bccd33df9f2e4099312543d65b6198633c6c3/Classes/Finishers/RedirectFinisher.php#L58
$request->getHttpRequest()->getBaseUri() has a trailing slash and $uri has a leading slash.

As a workaround I changed the form-builder fusion to generate an absolute uri for the redirect:
```
prototype(Neos.Form.Builder:NodeBasedFinisherCollection) {
    itemRenderer {
        redirectFinisher {
            renderer {
                element.options {
                    uri.@process.convertNodeUris.absolute = ${true}
                }
            }
        }
    }
}
```
But I think the code, that prepends scheme and host to the uri should be fixed in the form package.

### PR DESCRIPTION
**Pull request by ger4003**
This PR is pointed to 4.2 branch and makes #99 obsolete.

Closes  #89

### CODE REVIEW

**Comment by ger4003** [comment]
Hey @kitsunet ,

since you had a look into the code already could you review this change?

File: Classes/Finishers/RedirectFinisher.php [hunk 1]
```diff
@@ -55,7 +57,11 @@ public function executeInternal()
 
         $uriParts = parse_url($uri);
         if (!isset($uriParts['scheme']) || $uriParts['scheme'] === '') {
-            $uri = $request->getHttpRequest()->getBaseUri() . $uri;
+            $uri = sprintf(
+                '%s/%s',
+                rtrim($request->getHttpRequest()->getAttribute(Request::ATTRIBUTE_BASE_URI), '/'),
```
Comment (albe) [inline Classes/Finishers/RedirectFinisher.php]: Any reason why `$request->getHttpRequest()->getBaseUri()` got replaced with `$request->getHttpRequest()->getAttribute(Request::ATTRIBUTE_BASE_URI)`? The former was removed with Flow 6.0, but as Christian mentioned that should not be necessary for that version, so we need to carefully adjust this when upmerging.

Comment (ger4003) [reply Classes/Finishers/RedirectFinisher.php]: Not really. I will change it back so it does not break when using neos/flow < 5.1

**Comment by ger4003** [comment]
Hi @albe ,

can you initiate a new release for 4.2 branch including this PR?

Thanks.

**Comment by albe** [comment]
Sorry, totally forgot about this. Have upmerged and tagged releases now, so should be available from packagist soon. Sorry again for the delay!

**Comment by mgrundkoetter** [comment]
Now the package is broken:

> syntax error, unexpected ‘,’
> Type: ParseError
>   File:
> Packages/Application/Neos.Form/Classes/Finishers/RedirectFinisher.php
>   Line: 79

Happened in neos/form: 5.0.3 with PHP 7.2

**Comment by albe** [comment]
Yeah, sorry, that was a manual upmerge error by me :( Already fixed it yesterday with a new release 5.0.4

**Pull request merged**